specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,1006 @@
+attached to the entry point. The value of the attribute must be the string
+representation of a non-negative 64-bit integer constant.
+
+Similarly, the number of measurement results that need to be stored when
+executing the entry point function is captured by the `"required_num_results"`
+attribute.
+
+Beyond the entry point-specific requirements related to attributes, custom
+attributes may optionally be attached to any of the declared functions. The
+`irreversible` attribute in particular impacts how the program logic in the
+entry point is structured, as described in the section about the [entry point
+definition](#entry-point-definition). Furthermore, the following [LLVM
+attributes](https://llvm.org/docs/LangRef.html#function-attributes) may be used
+according to their intended purpose on function declarations and call sites:
+`inlinehint`, `nofree`, `norecurse`, `readnone`, `readonly`, `writeonly`, and
+`argmemonly`.
+
+## Module Flags Metadata
+
+The following [module
+flags](https://llvm.org/docs/LangRef.html#module-flags-metadata) must be added
+to the QIR bitcode:
+
+- a flag with the string identifier `"qir_major_version"` that contains a
+  constant value of type `i32`
+- a flag with the string identifier `"qir_minor_version"` that contains a
+  constant value of type `i32`
+- a flag with the string identifier `"dynamic_qubit_management"` that contains a
+  constant `true` or `false` value of type `i1`
+- a flag with the string identifier `"dynamic_result_management"` that contains
+  a constant `true` or `false` value of type `i1`
+
+The following [module
+flags](https://llvm.org/docs/LangRef.html#module-flags-metadata) may be added to
+the QIR bitcode to indicate support/use of optional capabilities. A lack of
+these module flags indicates that these capabilities are not used in the
+program.
+
+- a flag with the string identifier `"int_computations"` that contains a string
+  value where the string value is a comma-separated list of the supported/used
+  integer precision(s). For example, `!0 = !{i32 5, !"int_computations",
+  !"i32,i64"}`. Classical computations on integers of all listed precisions must
+  be supported by the executing backend. An empty value indicates that no
+  integer computations are supported/used.
+- a flag with the string identifier `"float_computations"` that contains a
+  string value where the string value is a comma-separated list of the
+  supported/used floating-point precision(s). For example, `!0 = !{i32 5,
+  !"float_computations", !"f32,f64"}`. The precision must be one of the LLVM
+  recognized values (f16, f32, f64, f80, or f128), and classical computations on
+  floating point numbers of all listed precisions must be supported by the
+  executing backend. An empty value indicates that no floating-point
+  computations are supported/used.
+- a flag with the string identifier `"fixedpoint_computations"` that contains a
```

Comment:
This point should be removed since fixed point computations are no longer part of the adaptive profile spec.